### PR TITLE
QA-12920 : use the new field existsInLive to display deletePermanentl…

### PR DESCRIPTION
**Pull request by Yousria**
…y or PublishDeletion actions

## JIRA

https://jira.jahia.org/browse/QA-12920

### CODE REVIEW

File: src/javascript/JContent/actions/deletePermanentlyAction.jsx [hunk 1]
```diff
@@ -7,7 +7,8 @@ import {PATH_CONTENTS_ITSELF, PATH_FILES_ITSELF} from './actions.constants';
 
 const checkAction = node => node.operationsSupport.markForDeletion &&
     isMarkedForDeletion(node) &&
-    node.aggregatedPublicationInfo.publicationStatus === 'NOT_PUBLISHED';
+    node.aggregatedPublicationInfo.publicationStatus === 'NOT_PUBLISHED' &&
+    node.aggregatedPublicationInfo.existsInLive !== undefined && !node.aggregatedPublicationInfo.existsInLive;
```
Comment (cedricmoitrier) [inline src/javascript/JContent/actions/deletePermanentlyAction.jsx:11]: I don't think this works as expected. As soon as existsInLive is defined then the whole expression will always resolve to false, right?

The green part should be something like:
`publicationStatus === 'NOT_PUBLISHED' && (existsInLive === undefined) ? true : !existsInLive`

I've omitted the 'node.aggregatedPublicationInfo' parts for the sake of clarification here.

File: src/javascript/JContent/actions/publishDeletionAction.jsx [hunk 2]
```diff
@@ -6,7 +6,8 @@ import {useSelector} from 'react-redux';
 
 const checkAction = node => node.operationsSupport.publication &&
     isMarkedForDeletion(node) &&
-    node.aggregatedPublicationInfo.publicationStatus !== 'NOT_PUBLISHED';
+    (node.aggregatedPublicationInfo.publicationStatus !== 'NOT_PUBLISHED' ||
+        (node.aggregatedPublicationInfo.publicationStatus === 'NOT_PUBLISHED' && node.aggregatedPublicationInfo.existsInLive));
```
Comment (cedricmoitrier) [inline src/javascript/JContent/actions/publishDeletionAction.jsx:10]: You should check whether or not existsInLive is undefined here as well.

File: src/javascript/JContent/actions/deletePermanentlyAction.jsx (hunk 1 above)
Comment (Yousria) [reply src/javascript/JContent/actions/deletePermanentlyAction.jsx:11]: you're right we should keep the old behaviour

File: src/javascript/JContent/actions/publishDeletionAction.jsx (hunk 2 above)
Comment (Yousria) [reply src/javascript/JContent/actions/publishDeletionAction.jsx]: done

File: src/javascript/JContent/JContent.utils.js [hunk 3]
```diff
@@ -124,6 +124,19 @@ function uppercaseFirst(string) {
     return string.charAt(0).toUpperCase() + string.substr(1);
 }
 
+const findField = (fieldName, types) => {
```
Comment (cedricmoitrier) [inline src/javascript/JContent/JContent.utils.js:127]: I may missing something but how do you know that fieldName exists in the expected type? From my understanding this method will return true if fieldName exists in any type.

Comment (cedricmoitrier) [reply src/javascript/JContent/JContent.utils.js:127]: Don't we need to either check the type in this method, or parameterize the hook to only return fields from a given type?

File: src/javascript/JContent/actions/deletePermanentlyAction.jsx [hunk 4]
```diff
@@ -1,23 +1,30 @@
...
     isMarkedForDeletion(node) &&
-    node.aggregatedPublicationInfo.publicationStatus === 'NOT_PUBLISHED';
+    node.aggregatedPublicationInfo.publicationStatus === 'NOT_PUBLISHED' &&
+    (node.aggregatedPublicationInfo.existsInLive === undefined ? true : !node.aggregatedPublicationInfo.existsInLive);
 
 export const DeletePermanentlyActionComponent = ({context, render: Render, loading: Loading}) => {
     const {language} = useSelector(state => ({language: state.language}));
 
+    let types = useSchemaFields();
+    let existsInLiveField = types === undefined ? false : findField('existsInLive', types);
```
Comment (tdraier) [inline src/javascript/JContent/actions/deletePermanentlyAction.jsx:19]: This code should be done directly by useNodeInfo

File: src/javascript/JContent/JContent.utils.js (hunk 3 above)
Comment (tdraier) [reply src/javascript/JContent/JContent.utils.js:127]: yes, agree - you should always ask for a field in a specific type , not look in all types

File: src/javascript/JContent/actions/deletePermanentlyAction.jsx (hunk 4 above)
Comment (Yousria) [reply src/javascript/JContent/actions/deletePermanentlyAction.jsx:19]: Ok

**Pull request merged**